File: Aula_15/ex01.cpp
#include <iostream>
#include <chrono>

int main() {
    // Número de operações
    const int n = 1000000;

    // Iniciar o cronômetro
    auto start = std::chrono::high_resolution_clock::now();

    // Realizar operações de ponto flutuante
    for (int i = 0; i < n; ++i) {
        volatile double result = 3.14159 * 2.71828;  // Exemplo de operação
    }

    // Parar o cronômetro
    auto end = std::chrono::high_resolution_clock::now();

    // Calcular o tempo decorrido
    std::chrono::duration<double> duration = end - start;
    
    // Calcular operações por segundo
    double operations_per_second = n / duration.count();

    std::cout << "Operações por segundo: " << operations_per_second << std::endl;

    return 0;
}

File: Aula_15/ex01.py
import time

# Número de operações
n = 1000000

# Medir o tempo de execução de operações de ponto flutuante
start_time = time.time()

# Realizar operações de ponto flutuante
for i in range(n):
    _ = 3.14159 * 2.71828  # Exemplo de operação

end_time = time.time()

# Calcular operações por segundo
elapsed_time = end_time - start_time
operations_per_second = n / elapsed_time

print(f"Operações por segundo: {operations_per_second:.2f}")
